cmd/operator/internal/controller/clusterkubearchiveconfig_controller.go

```diff
@@ -0,0 +1,128 @@
+// k9e  => shorthand for KubeArchive
+
+import (
+	"context"
+
+	rbacv1 "k8s.io/api/rbac/v1"
+
+	meta "k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	kubearchivev1alpha1 "github.com/kubearchive/kubearchive/cmd/operator/api/v1alpha1"
+	"github.com/kubearchive/kubearchive/cmd/sink/filters"
+)
+
+// ClusterKubeArchiveConfigReconciler reconciles a ClusterKubeArchiveConfig object
+type ClusterKubeArchiveConfigReconciler struct {
+	Client client.Client
+	Scheme *runtime.Scheme
+	Mapper meta.RESTMapper
+}
+
+//+kubebuilder:rbac:groups=kubearchive.kubearchive.org,resources=clusterkubearchiveconfigs,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=kubearchive.kubearchive.org,resources=clusterkubearchiveconfigs/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=kubearchive.kubearchive.org,resources=clusterkubearchiveconfigs/finalizers,verbs=update
+//+kubebuilder:rbac:groups=core,resources=serviceaccounts,verbs=create;delete;get;list;update;watch
+//+kubebuilder:rbac:groups=rbac.authorization.k8s.io,resources=clusterroles;roles;rolebindings,verbs=bind;create;delete;escalate;get;list;update;watch
+//+kubebuilder:rbac:groups=sources.knative.dev,resources=apiserversources,verbs=create;delete;get;list;update;watch
+//+kubebuilder:rbac:groups="",resources=configmaps,verbs=create;delete;get;list;update;watch
+//+kubebuilder:rbac:groups="",resources=namespaces,verbs=get;list;update;watch
+
+func (r *ClusterKubeArchiveConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := log.FromContext(ctx)
+
+	log.Info("Reconciling ClusterKubeArchiveConfig")
+
+	ckaconfig := &kubearchivev1alpha1.ClusterKubeArchiveConfig{}
+	if err := r.Client.Get(ctx, req.NamespacedName, ckaconfig); err != nil {
+		// Ignore not-found errors, since they can't be fixed by an immediate requeue (we need
+		// to wait for a new notification), and we can get them on deleted requests.
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	finalizerName := "kubearchive.org/finalizer"
+
+	if ckaconfig.DeletionTimestamp.IsZero() {
+		// The object is not being deleted, add the finalizer if necessary.
+		if !controllerutil.ContainsFinalizer(ckaconfig, finalizerName) {
+			controllerutil.AddFinalizer(ckaconfig, finalizerName)
+			if err := r.Client.Update(ctx, ckaconfig); err != nil {
+				return ctrl.Result{}, err
+			}
+		}
+	} else {
+		// The object is being deleted.
+		if controllerutil.ContainsFinalizer(ckaconfig, finalizerName) {
+			// Finalizer is present, clean up filters from ConfigMap and remove Namespace label.
+
+			log.Info("Deleting ClusterKubeArchiveConfig")
+
+			// Clean filters.
+			err := deleteNamespaceFromFilterConfigMap(ctx, r.Client, filters.GlobalNamespace)
```

Comment:
Why does this code import filters from the sink only to use `GlobalNamespace` when this variable is already defined in the controller package in `cmd/operator/internal/controller/common.go`?

Reply:
It shouldn't.  We should use the imported value. I will remove the local declaration.

Reply:
Fixed.